subsys/suit/events/src/suit_events.c

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <suit_events.h>
+#include <zephyr/kernel.h>
+
+/* Helper mask, with 0th bits for all processors set. */
+#define ALL_CPUS_MASK 0x11111111
+
+/* Helper mask, with all event bits for a given processor set. */
+#define ALL_EVENT_MASK 0x0F
+
+/* The macimum CPU ID handled by this module. */
+#define CPU_N 8
+
+/* The number of bits per CPU inside the event mask. */
+#define BITS_N 4
+
+/* Internal event structure, common for all API calls. */
+static K_EVENT_DEFINE(suit_events);
+
+/* Convert (cpu_id, event mask) pair into global (common for all CPUs) event mask. */
+static uint32_t cpu_events_to_suit_events_mask(uint8_t cpu_id, uint8_t events)
+{
+	/* Check for valid CPU ID values. */
+	if ((cpu_id >= CPU_N) && (cpu_id != SUIT_EVENT_CPU_ID_ANY)) {
```

Comment:
Actually rather than doing `cpu_id >= CPU_N` wouldn't it be more generic to create a `CPU_SUPPORTED_MASK` which could be then compared as `(1U << cpu_id) & CPU_SUPPORTED_MASK` ? I assume there might be some platforms based on Haltium with some non-adjacent cpu ID values

Reply:
Since ATM CPU ID is always used as a bit index by this module, it supports values 0-4, so the check above is sufficient.
If we will need to support non-adjacent IDs, a translating function must be added and a different checks for the input value introduced.
I think we may postpone such changes until we will face such situation.